### PR TITLE
Add blake3 python package

### PR DESCRIPTION
**Pull request by maartenbreddels**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Original PR was #12703 which was closed & force pushed.

### CODE REVIEW

**Comment by xhochy** [comment]
This looks good but as with all rust-based packages we have the problem that we need to include their third-party licenses.

**Comment by maartenbreddels** [comment]
Could you elaborate, and/or have an example PR that I can copy?

File: recipes/blake3/library_licenses/BLAKE3-LICENSE-CC0 [hunk 1]
```diff
@@ -0,0 +1 @@
+404: Not Found
```
Comment (xhochy) [inline recipes/blake3/library_licenses/BLAKE3-LICENSE-CC0:1]: Can you fix this?

**Comment by maartenbreddels** [comment]
Note: https://github.com/conda-forge/staged-recipes/pull/14509 seemed to be a good template/example for this, including the https://github.com/cds-astro/get-license-helper tool by @fxpineau ❤️

File: recipes/blake3/library_licenses/BLAKE3-LICENSE-CC0 (hunk 1 above)
Comment (maartenbreddels) [reply recipes/blake3/library_licenses/BLAKE3-LICENSE-CC0]: done

**Comment by maartenbreddels** [comment]
Thanks @xhochy !

**Pull request merged**